File: server/Gate/Center/ServerInfoHandler.cs
//////////////////////////////////////////////////////////////////////////
// 
// 文件：server/Gate/Center/ServerInfoHandler.cs
// 作者：Xoen Xie
// 时间：2023/06/19
// 描述：服务器相关定义
// 说明：
//
//////////////////////////////////////////////////////////////////////////
using CC;
using XSF;

public class ServerInfoHandler : Singleton<ServerInfoHandler>, IServerInfoHandler
{
    // 有一个服务器连入到集群
    public void OnServerNew(ServerInfo info)
    {
        
    }

    // 有一个服务器断开
    public void OnServerLost(uint nID)
    {

    }

    public void OnServerOk(ServerInfo info)
    {
        var sid = ServerID.GetSID(info.ID);
        if(sid.Type == (byte)EP.Game)
        {
            ConnectorManager.Instance.CreateConnector(info.ID, info.IP, (int)info.Ports[(int)EP.Gate]);
        }
    }
}

File: server-cpp/source/gate/server/src/ServerInfoHandler.cpp
//////////////////////////////////////////////////////////////////////////
//
// 文件：source/gate/server/src/ServerInfoHandler.cpp
// 作者：Xoen Xie
// 时间：2020/10/12
// 描述：中心服处理器
// 说明：
//
//////////////////////////////////////////////////////////////////////////
#include "ServerInfoHandler.h"
#include "XSF.h"
using namespace xsf;
#include "ConnectorManager.h"

void ServerInfoHandler::OnServerNew(ServerInfo *pInfo)
{

}

void ServerInfoHandler::OnServerLost(uint nID)
{

}

void ServerInfoHandler::OnServerOk(ServerInfo *pInfo)
{
    SID sid;
    sid.ID = pInfo->ID;
    if(sid.S.type == EP_Game)
    {
        ConnectorManager::Instance()->CreateConnector(pInfo->ID, pInfo->sIP, pInfo->Ports[EP_Gate]);
    }
}
